compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -634,6 +634,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             };
         let is_method = mode == Mode::MethodCall;
         let unsatisfied_predicates = &no_match_data.unsatisfied_predicates;
+        if let Err(guar) = unsatisfied_predicates.error_reported() {
+            return guar;
+        }
+
```

Comment:
This is necessary to avoid bogus diagnostics like

```

error[E0599]: the method `my_index` exists for struct `Scope<_>`, but its trait bounds were not satisfied
  --> $DIR/ice-failed-to-resolve-instance-for-110696.rs:50:51
   |
LL | struct Scope<T>(Phantom2<DummyT<T>>);
   | --------------- method `my_index` not found for this struct because it doesn't satisfy `Scope<{type error}>: MyIndex<DummyT<{type error}>>`
...
LL |     let _pos: Phantom1<DummyT<()>> = Scope::new().my_index();
   |                                                   ^^^^^^^^ method cannot be called on `Scope<_>` due to unsatisfied trait bounds
   |
note: trait bound `{type error}: Sized` was not satisfied
  --> $DIR/ice-failed-to-resolve-instance-for-110696.rs:41:6
   |
LL | impl<T: MyFrom<Phantom2<DummyT<U>>>, U> MyIndex<DummyT<T>> for Scope<U> {
   |      ^                               ^  ------------------     --------
   |      |                               |
   |      |                               unsatisfied trait bound introduced here
   |      unsatisfied trait bound introduced here
   = help: items from traits can only be used if the trait is implemented and in scope
note: `MyIndex` defines an item `my_index`, perhaps you need to implement it
  --> $DIR/ice-failed-to-resolve-instance-for-110696.rs:9:1
   |
LL | trait MyIndex<T> {
   | ^^^^^^^^^^^^^^^^
help: consider relaxing the type parameter's implicit `Sized` bound
   |
LL | impl<T: ?Sized + MyFrom<Phantom2<DummyT<U>>>, U> MyIndex<DummyT<T>> for Scope<U> {
   |         ++++++++
help: consider relaxing the type parameter's implicit `Sized` bound
   |
LL | impl<T: MyFrom<Phantom2<DummyT<U>>>, U: ?Sized> MyIndex<DummyT<T>> for Scope<U> {
   |                                       ++++++++
```

Not sure yet how to generally avoid these (I don't want to just hide the predicate, I want to hide the entire error), but this change doesn't make things worse really, so 🤷